test/extended/cmd.sh

```diff
@@ -71,75 +72,69 @@ popd
 rm -rf $tmp
 
 # ensure a local-only image gets a docker image(not imagestream) reference created.
-[ "$(oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker -o yaml |& tr '\n' ' ' | grep -E "from:\s+kind:\s+DockerImage\s+name:\s+test/scratchimage:latest\s+")" ]
+os::cmd::expect_success_and_text "oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker -o yaml |& tr '\n' ' '" "from:\s+kind:\s+DockerImage\s+name:\s+test/scratchimage:latest\s+"
 # error due to partial match
-[ "$(oc new-app test/scratchimage2 -o yaml |& tr '\n' ' ' 2>&1 | grep -E "partial match")" ]
+os::cmd::expect_failure_and_text "oc new-app test/scratchimage2 -o yaml" "partial match"
 # success with exact match	
-[ "$(oc new-app test/scratchimage -o yaml)" ]
+os::cmd::expect_success "oc new-app test/scratchimage"
 echo "[INFO] newapp: ok"
 
 echo "[INFO] Running env variable expansion tests"
-oc new-project envtest
-oc create -f test/extended/fixtures/test-env-pod.json
-tryuntil "oc get pods | grep Running"
+os::cmd::expect_success 'oc new-project envtest'
+os::cmd::expect_success 'oc create -f test/extended/fixtures/test-env-pod.json'
+os::cmd::try_until_text 'oc get pods' 'Running'
 podname=$(oc get pods --template='{{with index .items 0}}{{.metadata.name}}{{end}}')
-oc exec test-pod env | grep podname=test-pod
-oc exec test-pod env | grep podname_composed=test-pod_composed
-oc exec test-pod env | grep var1=value1
-oc exec test-pod env | grep var2=value1
-oc exec test-pod ps ax | grep "sleep 120"
+os::cmd::expect_success_and_text 'oc exec test-pod env' 'podname=test-pod'
+os::cmd::expect_success_and_text 'oc exec test-pod env' 'podname_composed=test-pod_composed'
+os::cmd::expect_success_and_text 'oc exec test-pod env' 'var1=value1'
+os::cmd::expect_success_and_text 'oc exec test-pod env' 'var2=value1'
+os::cmd::expect_success_and_text 'oc exec test-pod ps ax' 'sleep 120'
 echo "[INFO] variable-expansion: ok"
 
 echo "[INFO] Running image pull secrets tests"
-oc login ${MASTER_ADDR} -u pull-secrets-user -p password --certificate-authority=${MASTER_CONFIG_DIR}/ca.crt
+VERBOSE=true os::cmd::expect_success "oc login ${MASTER_ADDR} -u pull-secrets-user -p password --certificate-authority=${MASTER_CONFIG_DIR}/ca.crt"
```

Comment:
I'll give you a chance to defend.  Why would I choose to do this instead of simply running the login?